suite/tests/linux/sigcontext.c

```diff
@@ -96,10 +96,9 @@ signal_handler(int sig, siginfo_t *siginfo, ucontext_t *ucxt)
              */
             kernel_xstate_t *xstate = (kernel_xstate_t *)ucxt->uc_mcontext.fpregs;
             if (xstate->fpstate.sw_reserved.magic1 == FP_XSTATE_MAGIC1) {
-                assert(xstate->fpstate.sw_reserved.xstate_size >= sizeof(*xstate));
-                /* we can't print b/c not all processors have avx */
+                assert(xstate->fpstate.sw_reserved.extended_size >= sizeof(*xstate));
```

Comment:
But extended_size includes the x87 state, doesn't it?  The original seems what we want.

Reply:
It's my understanding that the size is extended_size if magic1 == FP_XSTATE_MAGIC1. i.e. the legacy region that can be used for x87 + extensions. From the kernel sources:

```
[..]
If sw_reserved.magic1 == FP_XSTATE_MAGIC1 then there's a
 * sw_reserved.extended_size bytes large extended context area present.
[..]
```

This matched my observation, where with AVX (not even AVX-512 yet), xstate_size is < sizeof(xstate).

Reply:
It appears that newer kernel sources have clarified extended_size, and it seems to depict the extended xstate area's size as discussed offline:

```
* If sw_reserved.magic1 == FP_XSTATE_MAGIC1 then there's a
 * sw_reserved.extended_size bytes large extended context area present. (The
 * last 32-bit word of this extended area (at the
 * fpstate+extended_size-FP_XSTATE_MAGIC2_SIZE address) is set to
 * FP_XSTATE_MAGIC2 so that you can sanity check your size calculations.)
 *
 * This extended area typically grows with newer CPUs that have larger and
 * larger XSAVE areas.
```

I suggest updating our sigcontext.h file.

Reply:
Historically have we copied this file, and if so, from which version?

Reply:
OK.  Maybe augment the comment in our sigcontext.h to remind future-us?

Reply:
Resolving, have not updated the header file yet. Separately?

Reply:
> OK. Maybe augment the comment in our sigcontext.h to remind future-us?

Done (pulled in the kernel version of the comment).